job-server/test/spark.jobserver/JobInfoActorSpec.scala

```diff
@@ -93,12 +93,31 @@ with FunSpecLike with Matchers with BeforeAndAfter with BeforeAndAfterAll {
       val jobInfo2 = JobInfo("foo-2", "context", JarInfo("demo", dt2), "com.abc.meme", dt2, None, None)
       dao.saveJobInfo(jobInfo1)
       dao.saveJobInfo(jobInfo2)
-      actor ! GetJobStatuses(Some(1))
+      actor ! GetJobStatuses(Some(1), Some(""))
       expectMsg(Seq[JobInfo](jobInfo1))
     }
-
+    it("should return job infos as requested status ") {
+      val dt1 = DateTime.parse("2013-05-28T00Z")
+      val dt2 = DateTime.parse("2013-05-29T00Z")
+      val jarInfo = JarInfo("demo", dt1)
+      val someError =  Some(new Throwable("test-error"))
+      val runningJob = JobInfo("running-1", "context", jarInfo, "com.abc.meme", dt1, None, None)
+      val errorJob = JobInfo("error-1", "context", jarInfo, "com.abc.meme", dt1, None, someError)
+      val finishedJob = JobInfo("finished-1", "context", jarInfo, "com.abc.meme", dt1, Some(dt2), None)
+
+      dao.saveJobInfo(runningJob)
+      dao.saveJobInfo(errorJob)
+      dao.saveJobInfo(finishedJob)
+
+      actor ! GetJobStatuses(Some(1), Some(JobInfo.STATUS_RUNNING))
+      expectMsg(Seq[JobInfo](runningJob))
+      actor ! GetJobStatuses(Some(1), Some(JobInfo.STATUS_ERROR))
+      expectMsg(Seq[JobInfo](errorJob))
+      actor ! GetJobStatuses(Some(1), Some(JobInfo.STATUS_FINISHED))
+      expectMsg(Seq[JobInfo](finishedJob))
+    }
     it("should return empty list if jobs doest not exist") {
-      actor ! GetJobStatuses(Some(1))
+      actor ! GetJobStatuses(Some(1), Some(""))
```

Comment:
I have rename object JobInfo to JobStatus ,   but list all jobs is unnecesssary,I think.